src/app/login/login.component.ts

```diff
@@ -5,77 +5,135 @@ import { Router, ActivatedRoute } from '@angular/router';
 
 require('./login.scss');
 
+export class UserData  {
+  firstName: string;
+  middleName: string;
+  lastName: string;
+  login: string;
+  password: string;
+  repeatPassword: string;
+  email: string;
+  phone: number;
+  language: string;
+  gender: string;
+  levels: number;
+  birthday: number;
+  birthmonth: number;
+  birthyear: number;
+  community: string;
+  region: string;
+  nation: string;
+}
+
 @Component({
   templateUrl: './login.component.html',
   styleUrls: [ './login.scss' ],
   encapsulation: ViewEncapsulation.None
 })
-export class LoginComponent {
-  constructor(
-    private couchService: CouchService,
-    private router: Router,
-    private route: ActivatedRoute
-  ) { }
-
-  createMode: boolean = this.router.url.split('?')[0] === '/login/newuser';
-  returnUrl = this.route.snapshot.queryParams['returnUrl'] || '/';
-  model = { name: '', password: '', repeatPassword: '' };
-  message = '';
-
-  onSubmit() {
-    if (this.createMode) {
-      this.checkAdminExistence().then((noAdmin) => {
-        if (noAdmin) {
-          this.createAdmin(this.model);
-        } else {
-          this.createUser(this.model);
-        }
-      });
-    } else {
-      this.login(this.model);
-    }
-  }
 
-  reRoute() {
-    this.router.navigate([ this.returnUrl ]);
-  }
+export class LoginComponent implements OnInit {
+
+ newUser: UserData = new UserData();

```

Comment:
This is a little redundant since `newUser` will be of type `UserData` since you assigning it as a `new UserData()`